internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -121,6 +122,16 @@ func (r *ClusterDeploymentsReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 		return result, err
 	}
 
+	// check for install config overrides and update if needed
+	updated, result, err = r.updateInstallConfigOverrides(ctx, clusterDeployment, cluster)
+	if err != nil {
+		return r.updateState(ctx, clusterDeployment, cluster, err)
+	}
+
+	if updated {
+		return result, err
+	}
```

Comment:
I see that this code exists also above, but is it right?  Each reconcile only updates one thing and then we return?

Reply:
Yes, this is the idea, but i think that there is a bug with it. if something was updated then we need to reconcile gain, so need to return `requeue = true`

Reply:
If the resource gets saved, that generates an event and will induce reconciliation. It's normal for a controller to not return `requeue = true` in that circumstance. It probably wouldn't hurt, but waiting for the event gives time to ensure the local cache is updated also.

Reply:
Resolving, @avishayt feel free to reopen if you think it should do all updates at once (it used to...)

Reply:
I don't really understand the logic of returning early, but if that's the best practice...